version: '3.8'

services:
  influxdb:
    image: influxdb:latest
    container_name: influxdb
    ports:
      - "8086:8086"
    environment:
      - INFLUXDB_DB=mydatabase
      - INFLUXDB_ADMIN_USER=admin
      - INFLUXDB_ADMIN_PASSWORD=admin_password
    volumes:
      - influxdb_data:/var/lib/influxdb
    restart: unless-stopped

  grafana:
    image: grafana/grafana:latest
    container_name: grafana
    ports:
      - "3000:3000"
    environment:
      - GF_SECURITY_ADMIN_USER=admin
      - GF_SECURITY_ADMIN_PASSWORD=admin_password
    volumes:
      - grafana_data:/var/lib/grafana
    depends_on:
      - influxdb
    restart: unless-stopped

  sensor_app:
    build: ./sensor_app
    restart: always 
    ports:
      - "5003:80"
    volumes:
      - ./sensor_app:/app  # ahora le digo que use la carpeta donde tengo todo
    devices:
      - "/dev/gpiomem:/dev/gpiomem"
      - "/dev/mem:/dev/mem"
    privileged: true    
    depends_on:
      - influxdb 
    

  # sensor_app:
  #   build: ./sensor_app
  #   image: python:3.12 # python:3.9-slim
  #   container_name: sensor_app
  #   volumes:
  #     - .docker/app:/app
  #   working_dir: /app 
  #   command: ["python", "app.py"]
  #   depends_on:
  #     - influxdb
  #   restart:  unless-stopped

volumes:
  influxdb_data:
    driver: local
  grafana_data:
    driver: local
